js/blocks/controls/fetch-input.js

```diff
@@ -152,13 +150,34 @@ class FetchInput extends Component {
 	 * That has its own handler, which will eventually hide the Popover.
 	 */
 	onBlur( event ) {
-		if ( event.relatedTarget && ! event.relatedTarget.classList.contains( 'bl-fetch-input__suggestion' ) ) {
+		if (
+			event.relatedTarget &&
+			! event.relatedTarget.classList.contains( 'components-popover__content' ) &&
+			! event.relatedTarget.classList.contains( 'bl-fetch-input__suggestion' )
+		) {
 			this.setState( {
 				showSuggestions: false,
 			} );
+
+			if ( '' === this.props.value ) {
+				return;
+			}
+
+			const matchingResults = this.state.results.filter(
+				suggestions => ( suggestions.slug === this.props.value )
+			);
+
+			if ( ! matchingResults.length ) {
+				this.selectLink( this.state.results[ this.state.selectedSuggestion ] );
+			}
 		}
 	}
 
+	onFocus() {
+		const inputValue = this.props.value;
+		this.updateSuggestions( inputValue );
```

Comment:
Show the dropdown immediately when focusing on the field.

Reply:
Nice, that's a great improvement. People might not know the exact username they're looking for.